instance/models/load_balancer.py

```diff
@@ -111,6 +148,28 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.domain
 
+    def modifies_configuration(method):  # nopep8 # pylint: disable=no-self-argument
+        """
+        Flag an internal method as one that modifies the state of load balancer configuration,
+        to allow managing configuration state outside and around the method call.
+        """
+        @functools.wraps(method)
+        def wrapper(self, *args, **kwargs):
+            """
+            Used to wrap the function that modified load balancer configuration state.
+            """
+            with transaction.atomic():
+                self.configuration_version += 1
+                self.save()
```

Comment:
I'm not sure this ensures atomicity of the update, since `self` will already be loaded from the DB before the transaction even starts.  To be honest, I'm not even sure this _needs_ to be atomic, since if two processes update this field in parallel, we don't really care whether the value gets increased by one or by two – both would mark the LB config as dirty.

So this is kind of up to you.  If you think this needs to be atomic, make sure it is, e.g. by using an F-expression like

    from django.db.models import F
    self.configuration_version = F("configuration_version") + 1

(Not sure I got the details right.)  If you think it's fine if this isn't atomic, just remove the transaction context manager.

Reply:
I tried some things to see if I could keep it atomic, including with F-expressions, but it was just unnecessarily complicating things, and I'm thinking the same as you now -- that atomicity here doesn't seem to be needed. Only one thread will grab the lock, and actually, we want that thread to have the version provided in-memory through `self.configuration_version`, when it originally updated it, so removing the context manager and the `candidate_configuration_version` below and just using `self.configuration_version` should suffice.

Reply:
Please do not remove `candidate_configuration_version`.  Relying on `self.configuration_version` seems fragile, since we would rely on the value not being updated from the DB.  So if any of the methods involved in bulding the configuration calls `self.refresh_from_db()`, we would have a race condition again, so it seems more robust to keep the local variable.

Reply:
Ah, yeah, good point!